marquez_client/client.py

```diff
@@ -80,16 +97,204 @@ def list_namespaces(self, limit=None, offset=None):
+        if len(source_name) > _SOURCE_LENGTH_LIMIT:
+            raise ValueError('source_name length is {0}, must be <= {1}',
+                             len(source_name), _SOURCE_LENGTH_LIMIT)
+
+        if not connection_url:
+            raise ValueError('connection_url must not be None')
+
+        if not isinstance(source_type, SourceType):
+            raise ValueError(f'source_type must be instance of SourceType')
+
+        payload = {
+            'type': source_type.value,
+            'connectionUrl': connection_url
+        }
+
+        if description:
+            payload['description'] = description
+
+        return self._put(self._url('/sources/{0}', source_name),
+                         payload=payload)
+
+    def get_source(self, source_name):
+        if not source_name:
+            raise ValueError('source_name must not be None')
+        if len(source_name) > _SOURCE_LENGTH_LIMIT:
+            raise ValueError('source_name length is {0}, must be <= {1}',
+                             len(source_name), _SOURCE_LENGTH_LIMIT)
+
+        return self._get(self._url('/sources/{0}', source_name))
+
+    def list_sources(self, limit=None, offset=None):
+        return self._get(
+            self._url('/sources'),
+            params={
+                'limit': limit,
+                'offset': offset
+            }
+        )
+
+    # Datasets API
+    def create_dataset(self, namespace_name, dataset_name, dataset_type,
+                       physical_name, source_name,
+                       description=None, run_id=None,
+                       schema_location=None,
+                       fields=None, tags=None):
+        if not namespace_name:
+            raise ValueError('namespace_name must not be None')
+        if len(namespace_name) > _NAMESPACE_LENGTH_LIMIT:
+            raise ValueError('namespace_name length is {0}, must be <= {1}',
+                             len(namespace_name), _NAMESPACE_LENGTH_LIMIT)
+
+        if not dataset_name:
+            raise ValueError('dataset_name must not be None')
+        if len(dataset_name) > _DATASET_LENGTH_LIMIT:
+            raise ValueError('dataset_name length is {0}, must be <= {1}',
+                             len(dataset_name), _DATASET_LENGTH_LIMIT)
+
+        if not isinstance(dataset_type, DatasetType):
+            raise ValueError('dataset_type must be instance of DatasetType')
+        if dataset_type == DatasetType.STREAM and not schema_location:
+            raise ValueError('STREAM type datasets must have schema_location')
+
+        if not physical_name:
+            raise ValueError('physical_name must not be None')
+
+        if not source_name:
+            raise ValueError('source_name must not be None')
+        if len(source_name) > _SOURCE_LENGTH_LIMIT:
+            raise ValueError('source_name length is {0}, must be <= {1}',
+                             len(source_name), _SOURCE_LENGTH_LIMIT)
```

Comment:
This is a good example that could be made more compact by reusing a function.
```
checkLength('namespace_name', namespace_name)
...
checkPresent('physical_name', physical_name)
checkLength('source_name', source_name)
```

Reply:
+1 for these util methods!

Reply:
Add maybe to be clear on the type of length check, we can define the method: `checkMaxLength(str, max_int)`

Reply:
done.
multiple utility methods added.